astropy/utils/tests/test_data.py

```diff
@@ -1140,12 +1140,12 @@ def osraiser(dirnm, linkto, pkgname=None):
 def test_read_unicode(filename):
     contents = get_pkg_data_contents(os.path.join("data", filename), encoding="utf-8")
     assert isinstance(contents, str)
-    contents = contents.splitlines()[1]
+    contents = contents.splitlines()[0]
```

Comment:
There's only one line now:
```suggestion
```
Furthermore, it might be a good idea to parametrize this test over the encoding.

Reply:
This line isn't a no-op, it actually removes a trailing newline, but I refactored it for clarity.

> Furthermore, it might be a good idea to parametrize this test over the encoding.

how do you mean ? are you suggesting to replace the existing parametrization on concrete filenames ?

Reply:
I'm suggesting that it might be a good idea to parametrize the test both over filenames and over encodings.

Reply:
Sorry I don't get what you mean. The encoding isn't explicitly referenced anywhere in the test so I don't see how that'd work.

Reply:
```python
@pytest.mark.parametrize(
    "filename",
    [
        ...
    ],
)
@pytest.mark.parametrize(
    "encoding,expected_type,expected_contents",
    [
        ...
    ],
)
def test_read_unicode(filename, encoding, expected_type, expected_contents):
    contents = get_pkg_data_contents(os.path.join("data", filename), encoding=encoding)
    assert type(contents) is expected_type
    assert contents == expected_contents
```

Reply:
Oh my. I think my brain was reading "compression algorithm" instead of "encoding" 🤦🏻‍♂️ 
Good idea indeed, but slightly out of scope I think. Let me try that in a follow up PR.

Reply:
#16538